src/afl-fuzz-state.c

```diff
@@ -102,7 +102,7 @@ void afl_state_init(afl_state_t *afl, uint32_t map_size) {
   afl->stats_update_freq = 1;
   afl->stats_file_update_freq_msecs = STATS_UPDATE_SEC * 1000;
   afl->stats_avg_exec = 0;
-  afl->skip_deterministic = 1;
```

Comment:
So this changes the default to not skip deterministic stage? Is this a good idea? If so, we need to update docs, right?

Reply:
this is what this PR is about. it is a new approach and makes afl++ slightly better. yes the documentation then needs to be updated.

Reply:
We replace the original deterministic stage to `skipdet`, which only do much less deterministic mutation. Therefore, this change means we use `skipdet` by default and original havoc only if `-d` is set.

Reply:
This is an evaluation we ran on fuzzbench. skipdet (aflplusplus_sk) show some improvement over AFL++ so I open this PR. https://fuzzbench.com/reports/experimental/2024-01-24-aflpp-sk2/index.html.